WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/ranges/data/AnalyticsHubLastMonthRangeData.kt

```diff
@@ -0,0 +1,42 @@
+package com.woocommerce.android.ui.analytics.ranges.data
+
+import com.woocommerce.android.extensions.endOfCurrentMonth
+import com.woocommerce.android.extensions.oneMonthAgo
+import com.woocommerce.android.extensions.startOfCurrentMonth
+import com.woocommerce.android.ui.analytics.ranges.AnalyticsHubTimeRange
+import com.woocommerce.android.ui.analytics.ranges.AnalyticsHubTimeRangeData
+import java.util.Calendar
+import java.util.Date
+
+// Responsible for defining two ranges of data, the current one, starting from the first day of the last month
+// until the final day of that same month, and the previous one as two months ago, also starting
+// from the first day until the final day of that month. E. g.
+//
+// Today: 29 Jul 2022
+// When user requests report at 05:49 PM
+// Current range: Jun 1, 00:00 until Jun 30, 23:59:59, 2022
+// Previous range: May 1, 00:00 until May 31, 23:59:59, 2022
+//
+class AnalyticsHubLastMonthRangeData(
```

Comment:
Nitpick: I personally would prefer to name all of those classes without `AnalyticsHub` prefix. Without it, it'd be easier to read the name and we know that this is `AnalyticsHub` related because of parent class. But please mind it's just my preference - if you prefer the prefix I totally get it 👍 

```suggestion
class LastMonthRangeData(
```

Reply:
Yeah, sounds redundant indeed! Thanks for the suggestion! Adjusted here: https://github.com/woocommerce/woocommerce-android/pull/8063/commits/87a70de5cdaa8b36b209d1ec4807e3f5efbc3ee5